src/components/snapshot/SnapshotPopup.tsx

```
@@ -0,0 +1,247 @@
+    },
+    bodyText: {
+      paddingBottom: theme.spacing(2),
+    },
+    light: {
+      color: 'rgba(0, 0, 0, 0.54)',
+    },
+    chip: {
+      backgroundColor: theme.palette.primary.light,
+      margin: theme.spacing(0.5),
+      marginLeft: '0px',
+    },
+    centered: {
+      textAlign: 'center',
+    },
+    exitButton: {
+      position: 'absolute',
+      top: 0,
+      right: 0,
+      padding: '1em',
+      width: '3em',
+      height: '3em',
+      cursor: 'pointer',
+      ...theme.mixins.jadeLink,
+    },
+    title: {
+      marginTop: '10px',
+      marginRight: '16px',
+      paddingBottom: '15px',
+    },
+    jadeLink: {
+      ...theme.mixins.jadeLink,
+    },
+  });
+
+interface SnapshotPopupProps {
+  classes: ClassNameMap;
+  dispatch: Dispatch<Action>;
+  filterData: any;
+  isOpen: boolean;
+  policies: Array<PolicyModel>;
+  snapshot: SnapshotModel;
+}
+
+function SnapshotPopup({
+  classes,
+  dispatch,
+  filterData,
+  isOpen,
+  policies,
+  snapshot,
+}: SnapshotPopupProps) {
+  /**
+   * When a snapshot gets created, the empty snapshot prop gets populated with a brief summary.
+   * We want to use that snapshot id to get the full object and the policies for the dialog.
+   */
+  const prevSnapshotRef = useRef<SnapshotModel>(snapshot);
+  useEffect(() => {
+    if (prevSnapshotRef.current && _.isEmpty(prevSnapshotRef.current) && !_.isEmpty(snapshot)) {
+      prevSnapshotRef.current = snapshot;
+      dispatch(
+        getSnapshotById({
+          snapshotId: snapshot.id,
+          include: [
+            SnapshotIncludeOptions.SOURCES,
+            SnapshotIncludeOptions.TABLES,
+            SnapshotIncludeOptions.RELATIONSHIPS,
+            SnapshotIncludeOptions.ACCESS_INFORMATION,
+            SnapshotIncludeOptions.PROFILE,
+            SnapshotIncludeOptions.DATA_PROJECT,
+          ],
+        }),
+      );
+      dispatch(getSnapshotPolicy(snapshot.id));
+    }
+  }, [snapshot, dispatch]);
```

Do we want to subscribe on `dispatch` here? I know we ran into some issues with that elsewhere

---

When I remove it, I get a missing dependency eslint error. I could add a comment to suppress the error, but that seems maybe unnecessary when it works correctly with dispatch in the dependency array. Is there a reason we would want to do this given there is not an issue with it here? Like is it a principle that we don't want to subscribe to dispatch?

---

I believe that adding the comment to suppress the error is the standard practice because we've run into issues with it making a large number of api calls if it triggers whenever dispatch updates. Because we don't have polling it might be okay?